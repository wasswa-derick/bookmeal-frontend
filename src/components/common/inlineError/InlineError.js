import React from "react";
import PropTypes from "prop-types";

const InlineError = ({ text }) => (
  <div className="invalid-feedback">{text}</div>
);

InlineError.propTypes = {
  text: PropTypes.string.isRequired
};

export default InlineError;
