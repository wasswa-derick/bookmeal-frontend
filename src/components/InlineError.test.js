import React from "react";
import { shallow } from "enzyme";
import InlineError from "./InlineError";

describe("InlineError", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<InlineError text="this is an error message" />);
  });

  it("should render the error message", () => {
    const errorMsg = wrapper.find(".invalid-feedback").text();
    expect(errorMsg).toEqual("this is an error message");
  });
});
