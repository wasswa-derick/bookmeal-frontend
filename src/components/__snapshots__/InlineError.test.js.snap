// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineError should render correctly 1`] = `
<div
  className="invalid-feedback"
>
  this is an error message
</div>
`;
